dotnet/sdk-phase-1.md

```diff
@@ -0,0 +1,1052 @@
+* All activities must have a non-inheritable `Activity` attribute
+  * Optional `Name` positional property can be given. Default is unqualified method name with "Async" trimmed off the
+    end if it returns a `Task`
+* Activity classes can be/implement interfaces
+  * There is no value to this from a Temporal perspective currently since the caller side of activities is not in this
+    phase
+* Activities do not have to be async, they will use the global thread pool otherwise (see worker later)
+* Non-static activity classes are encouraged to set a `static` `readonly` `Ref` property for use by clients
+  * Similar approach as workflows
+  * Unlike workflow validation that checks every public method attribute, this only ensures there is at least one public
+    activity
+  * Not required for static activities because they can just be referenced on the class
+* Any overridden instance activities from base classes or same-named static activities from base classes must have the
+  exact same signature and attribute. This ensures people don't make mistakes accidentally overriding and removing the
+  activity attribute. This also avoids a diamond problem or duplicate problem for statics.
+
+### Activity Contextual Operations
+
+The activity context looks like this:
+
+```csharp
+namespace Temporal.Activity;
+
+public class ActivityContext
+{
+    public static ActivityContext Current { get; }
+
+    public static bool HasCurrent { get; }
+
+    internal ActivityContext(
+        ActivityInfo info,
+        CancellationToken cancellationToken,
+        Action<object[]> heartbeater) { /*...*/ }
+
+    public ActivityInfo Info { get; }
+
+    public CancellationToken CancellationToken { get; }
+
+    public CancellationToken WorkerShutdownToken { get; }
+
+    internal Action<object[]> Heartbeater { get; }
+
+    public void Heartbeat(params object[] details) { /*...*/ }
+}
+```
+
+Notes:
+
+* The current activity context is on thread local or async local depending on whether activity is async or not
+* All calls except `HasCurrent` fail if not in activity context
+* `ActivityInfo` is an immutable record containing info
+* `CancellationToken` and `WorkerShutdownToken` are usable by sync and async
+  * `WorkerShutdownToken` is set on worker shutdown before cancellation token is set. If there is no graceful shutdown
+    period, one is set immediately after the other
+* Note the lack of logger here. See "Logging" later, but in general .NET doesn't have a logger in the standard library.
+  We may make some kind of `IReadOnlyDictionary<string, string> ActivityInfo.LogInfo { get; }` that can be used with
+  `ILogger.BeginScope` or similar.
+
+### Async Completion
+
+Simply `throw new Temporal.Activity.CompleteAsyncException` from the activity.
+
+## Client
+
+Inside the `Temporal.Client` namespace, we have the following types:
+
+* `TemporalClient` - namespace-specific, stateless high-level client to Temporal
+  * See later for most of its API design
```

Comment:
Would this client include other server APIs like `Schedules`?

Reply:
Yes